### ISSUE TITLE
digital ocean module does not wait for server to be actually deleted (absent)

### ISSUE THREAD

**Issue by elnygren** [ansible/ansible-modules-core#2402]
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

digital_ocean module
##### ANSIBLE VERSION

N/A
##### SUMMARY

Problem:
1. use digital_ocean module to destroy a server
2. module exits and says the server is destroyed
3. start a new server (same hostname, unique hostnames true)
4. the new server is actually the same server (and not a fresh clone of the OS template)

It's a bit nasty to receive the old server with the state it had when you destroyed it rather than a fresh OS. I noticed this problem while doing IaaS benchmarks with Ansible - the `apt-get update` and `apt-get install ...` phase for DO would suddenly be completed in just a couple seconds, implying that everything was already in place. 

Solution:

The digital_ocean module should wait for the server to be actually destroyed. I presume this must be done by polling the digital ocean API.

**Comment by robynbergeron** [comment]
@elnygren -- Thanks for submitting this bug report. You might want to consider giving us the following additional information to help the module maintainer identify the issue: https://github.com/ansible/ansible/blob/devel/ISSUE_TEMPLATE.md

@zbal --  thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module. 

Also pinging @skottler.

**Comment by ansibot** [comment]
@alukovenko, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by alukovenko** [comment]
Will be fixed in #4219, wontfix
